pkg/monitor/lease_test.go

```diff
@@ -0,0 +1,126 @@
+package monitor
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/milosgajdos/tenus"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/baremetal-runtimecfg/pkg/config"
+)
+
+var _ = Describe("lease_vip", func() {
+	var (
+		realIP    net.IP
+		realIface net.Interface
+		testIface *net.Interface
+		testIP    net.IP
+		testName  string
+		err       error
+	)
+
+	BeforeEach(func() {
+		host, _ := os.Hostname()
+		addrs, _ := net.LookupIP(host)
+		for _, addr := range addrs {
+			if ipv4 := addr.To4(); ipv4 != nil {
+				realIP = ipv4
+
+				testAddr := append(addr[:15], (addr[15] + 1))
+				testIP = testAddr.To4()
+
+				realIface, _, err = config.GetInterfaceAndNonVIPAddr([]net.IP{realIP})
+				Expect(err).ShouldNot(HaveOccurred())
+			}
+		}
+
+		testName = uuid.New().String()[:4]
+
+		printIPs()
+		printMACs()
+	})
+
+	Describe("createVIPInterface", func() {
+		It("happy_flow", func() {
+			testIface, err = createVIPInterface(testName, testIP)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("different_names", func() {
+			testIface, err = createVIPInterface(testName, testIP)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			iface2, err := createVIPInterface(uuid.New().String()[:4], testIP)
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(iface2.Name).ShouldNot(Equal(realIface.Name))
+			Expect(iface2.Name).ShouldNot(Equal(testIface.Name))
+			Expect(iface2.HardwareAddr).ShouldNot(Equal(realIface.HardwareAddr))
+			Expect(iface2.HardwareAddr).ShouldNot(Equal(testIface.HardwareAddr))
+
+			Expect(tenus.DeleteLink(iface2.Name)).ShouldNot(HaveOccurred())
+		})
+
+		It("same_names", func() {
+			testIface, err = createVIPInterface(testName, testIP)
+			Expect(err).ShouldNot(HaveOccurred())
+
+			_, err = createVIPInterface(testName, testIP)
+			Expect(err).Should(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			Expect(testIface.Name).ShouldNot(Equal(realIface.Name))
+			Expect(testIface.HardwareAddr).ShouldNot(Equal(realIface.HardwareAddr))
+
+			printIPs()
```

Comment:
What's the purpose of this printing? If we want to test that we created the interface as intended, we should use the net or the netlink golang packages to check it, right?

Reply:
You are correct. I added it just for self clarification of what's going on as debug printings.